driver/optmaindriver.ml

```diff
@@ -168,29 +166,31 @@ let main unix argv ppf ~flambda2 =
         let precision = !Clflags.timings_precision in
         let w2b n = n * (Sys.word_size / 8) in
         let fw2b x = w2b (Float.to_int x) in
-        Format.fprintf Format.std_formatter "%0.*fs gc\n" precision (secs (minor +. major));
-        Format.fprintf Format.std_formatter "  %0.*fs minor\n" precision (secs minor);
-        Format.fprintf Format.std_formatter "  %0.*fs major\n" precision (secs major);
-        Format.fprintf Format.std_formatter "- heap\n";
+        Format.fprintf ppf "%0.*fs gc\n" precision (secs (minor +. major));
+        Format.fprintf ppf "  %0.*fs minor\n" precision (secs minor);
+        Format.fprintf ppf "  %0.*fs major\n" precision (secs major);
+        Format.fprintf ppf "- heap\n";
         (* Having minor + major + promoted = total alloc make more sense for
-           hierarchical stats. *)
-        Format.fprintf Format.std_formatter "  %ib alloc\n"
+          hierarchical stats. *)
+        Format.fprintf ppf "  %ib alloc\n"
           (fw2b stats.minor_words + (fw2b stats.major_words - fw2b stats.promoted_words));
-        Format.fprintf Format.std_formatter "    %ib minor\n"
+        Format.fprintf ppf "    %ib minor\n"
           (fw2b stats.minor_words - fw2b stats.promoted_words);
-        Format.fprintf Format.std_formatter "    %ib major\n"
+        Format.fprintf ppf "    %ib major\n"
           (fw2b stats.major_words - fw2b stats.promoted_words);
-        Format.fprintf Format.std_formatter "    %ib promoted\n"
+        Format.fprintf ppf "    %ib promoted\n"
           (fw2b stats.promoted_words);
-        Format.fprintf Format.std_formatter "  %ib top\n" (w2b stats.top_heap_words);
-        Format.fprintf Format.std_formatter "  %i collections\n"
+        Format.fprintf ppf "  %ib top\n" (w2b stats.top_heap_words);
+        Format.fprintf ppf "  %i collections\n"
           (stats.minor_collections + stats.major_collections);
-        Format.fprintf Format.std_formatter "    %i minor\n" stats.minor_collections;
-        Format.fprintf Format.std_formatter "    %i major\n" stats.major_collections;
+        Format.fprintf ppf "    %i minor\n" stats.minor_collections;
+        Format.fprintf ppf "    %i major\n" stats.major_collections;
       end;
-      Profile.print
-        Format.std_formatter
-        !Clflags.profile_columns
-        ~timings_precision:!Clflags.timings_precision;
-    end;
+      Profile.print ppf !Clflags.profile_columns ~timings_precision:!Clflags.timings_precision
+    in
+    (* If -dump-into-csv is set, do not dump profile information to .dump file or stdout *)
```

Comment:
This comment falls into the trap of stating a fact but not indicating why the fact is important or interesting. I might suggest:
```suggestion
    (* -dump-to-csv means we should always produce a .csv file. *)
```
But this comment is maybe not even worth mentioning, and we should perhaps delete it instead.

Reply:
It seems that the code already conveys the functionality better than a comment would so I have removed it.